File: src/test/java/com/uapp_llc/test/comparator/TaskComparator.java
package com.uapp_llc.test.comparator;

import java.util.Comparator;

import com.google.common.collect.ComparisonChain;

import com.uapp_llc.model.Column;
import com.uapp_llc.model.Task;

class TaskComparator implements Comparator<Task> {

  private final Comparator<Column> columnComparator;

  TaskComparator(Comparator<Column> columnComparator) {
    this.columnComparator = columnComparator;
  }

  @Override
  public int compare(Task left, Task right) {
    return ComparisonChain.start()
        .compare(left.getId(), right.getId())
        .compare(left.getCreatedAt(), right.getCreatedAt(), NotNullComparator.leftNotNull())
        .compare(left.getName(), right.getName())
        .compare(left.getDescription(), right.getDescription())
        .compare(left.getColumn(), right.getColumn(), columnComparator)
        .result();
  }

}

File: src/test/kotlin/com/uapp_llc/test/comparator/TaskComparator.kt
package com.uapp_llc.test.comparator

import com.google.common.collect.ComparisonChain
import com.uapp_llc.model.Column
import com.uapp_llc.model.Task
import java.util.Comparator

class TaskComparator(

    private val columnComparator: Comparator<in Column>

) : Comparator<Task> {

  override fun compare(left: Task?, right: Task?): Int {
    return ComparisonChain.start()
        .compare(left?.id as Comparable<*>, right?.id as Comparable<*>)
        .compare(left.createdAt, right.createdAt, NotNullComparator.leftNotNull())
        .compare(left.name as Comparable<*>, right.name as Comparable<*>)
        .compare(left.description as Comparable<*>, right.description as Comparable<*>)
        .compare(left.index, right.index)
        .compare(left.column, right.column, columnComparator)
        .result()
  }

}
